### PR TITLE
New MIM JSON for LogCollector, Configuration, DO and ADHS (former DeviceHealthTelemetry)

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

**Comment by MatthewMWR** [comment]
Looks good overall, with a few minor questions
1. Should we delay changing protocol to ioTHubProtocol, pending updates of actual config file, tests, docs?
2. Should we use a more specific filename than "configuration.json"?

**Comment by MatthewMWR** [comment]
Looks good overall, with a few minor questions
1. Should we delay changing protocol to ioTHubProtocol, pending updates of actual config file, tests, docs?
2. Should we use a more specific filename than "configuration.json"?

File: src/modules/mim/configuration.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                  "name": "mqtt",
+                  "enumValue": 0
+                },
+                {
+                  "name": "mqttWebSocket",
+                  "enumValue": 1
+                }
+              ]
+            }
+          ]
```
Comment (robertschaedler3) [inline src/modules/mim/configuration.json:101]: Typo: extra `]`

Comment (MariusNi) [reply src/modules/mim/configuration.json]: Well spotted, thanks. Fixed!

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**